### ISSUE TITLE
[BUG] Imviz: WCS bounding boxes for NIRCam

### ISSUE THREAD

**Issue by hcferguson** [spacetelescope/jdaviz#1435]
I have what I think are  JWST NIRCam SW pipeline output data products -- but produced from simulations and not by me. So the funky behavior described here may be due to the data and not Imviz.

What I'm seeing is that the RA & Dec in the cursor readback vanishes when I get the cursor near the edge of one of the detectors, but while I'm still in a region that has data. So it's as if the WCS has some sort of bounding box in it that
describes the detector boundaries, and that bounding box is incorrect. 

**To Reproduce**
`imviz.load_data('ceers5_f150w_i2d_bkgsub.fits[SCI]')`
And watch the cursor readback near the boundaries of the detectors.

The data file is at https://drive.google.com/file/d/1uv-MxWCXFEBZNkglPy-jZWeLDqSuokKK/view?usp=sharing .

**Expected behavior**
Given that I read a fits SCI extension which seems to have a normal FITS WCS, I'm surprised that Imviz seems to have a bounding box. (I'm less surprised that it's incorrect, but it would be good to check that  -- if the bounding boxes are supposed to be there -- they are correct for real NIRCam pipeline output data products).

**Screenshots**
I've put a screen recording at https://drive.google.com/file/d/1uufbWVJVnO-SPg5hNmSdGuOH_4nyCeaK/view?usp=sharing .

**Desktop (please complete the following information):**

See #1411

**Package versions (please complete the following information):**

See #1411

**Comment by pllim** [comment]
If you inspect `imviz.app.data_collection[0].coords` , is it a `gwcs.WCS` or `astropy.wcs.WCS` object?

**Comment by pllim** [comment]
Also, if you zoom in more, does that weird behavior disappear? There is also a known issue about this:

* #964

**Comment by hcferguson** [comment]
It is a `gwcs.wcs.WCS` object. The behavior persists if I change the zoom.

**Comment by hcferguson** [comment]
Looks like the cursor readback is respecting the bounding box that is in this object:
```
imviz.app.data_collection[0].coords.bounding_box

ModelBoundingBox(
    intervals={
        x0: Interval(lower=0.0, upper=10250.175433948156)
        x1: Interval(lower=0.0, upper=4502.722279261141)
    }
    model=CompoundModel(inputs=('x0', 'x1'))
    order='C'
)
```

So the wcs is not  identifying the detector boundaries, but just an overall region in x,y that isn't quite right. Off more in the vertical direction (`x1` in the bounding box) than in the horizontal direction. Seems like this is a data issue and not an imviz issue. 

The fact that imviz is using the gwcs from the ASDF extension when it is available is probably desirable behavior.

**Comment by pllim** [comment]
Yes, I purposely have the parser give GWCS precedence because in some data, GWCS has distortion info that is not translated to the FITS header. I'll close out this issue then. Thanks for checking.
